src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -649,6 +671,92 @@ def filter(self, col: Expression) -> "SelectStatement":
 
         return new
 
+    def drop(
+        self, cols: List[Expression], kept_cols: List[Expression]
+    ) -> "SelectStatement":
+        disable_next_level_flatten = False
+        new_column_states = derive_column_states_from_subquery(kept_cols, self)
+        if new_column_states is None:
+            can_be_flattened = False
```

Comment:
We have a lot of conditions where `can_be_flattened` is `False`. I wonder if we should add a comment saying under what condition we will continue flattening the query.

Reply:
Added descriptions for situations where `exclude` variable is used; the other conditions mirror the conditions from `select()` in `select_statement.py` and have the same query flattening rationale.